litmus-portal/graphql-server/pkg/myhub/handler/handler.go

```diff
@@ -0,0 +1,153 @@
+	Kind        string             `yaml:"kind"`
+	Metadata    Metadata           `yaml:"metadata"`
+	Spec        Spec               `yaml:"spec"`
+	PackageInfo PackageInformation `yaml:"packageInfo"`
+	Experiments []Chart            `yaml:"experiments"`
+}
+
+//Maintainer ...
+type Maintainer struct {
+	Name  string
+	Email string
+}
+
+//Link ...
+type Link struct {
+	Name string
+	URL  string
+}
+
+//Metadata ...
+type Metadata struct {
+	Name        string     `yaml:"name"`
+	Version     string     `yaml:"version"`
+	Annotations Annotation `yaml:"annotations"`
+}
+
+//Annotation ...
+type Annotation struct {
+	Categories       string `yaml:"categories"`
+	Vendor           string `yaml:"vendor"`
+	CreatedAt        string `yaml:"createdAt"`
+	Repository       string `yaml:"repository"`
+	Support          string `yaml:"support"`
+	ChartDescription string `yaml:"chartDescription"`
+}
+
+//Spec ...
+type Spec struct {
+	DisplayName         string   `yaml:"displayName"`
+	CategoryDescription string   `yaml:"categoryDescription"`
+	Keywords            []string `yaml:"keywords"`
+
+	Maturity       string       `yaml:"maturity"`
+	Maintainers    []Maintainer `yaml:"maintainers"`
+	MinKubeVersion string       `yaml:"minKubeVersion"`
+	Provider       struct {
+		Name string `yaml:"name"`
+	} `yaml:"provider"`
+	Links []Link `yaml:"links"`
+
+	Experiments     []string `yaml:"experiments"`
+	ChaosExpCRDLink string   `yaml:"chaosexpcrdlink"`
+	Platforms       []string `yaml:"platforms"`
+	ChaosType       string   `yaml:"chaosType"`
+}
+
+//PackageInformation ...
+type PackageInformation struct {
+	PackageName string `yaml:"packageName"`
+	Experiments []struct {
+		Name string `yaml:"name"`
+		CSV  string `yaml:"CSV"`
+		Desc string `yaml:"desc"`
+	} `yaml:"experiments"`
+}
+
+//Charts ...
+type Charts []Chart
+
+//default path for storing local clones
+const (
+	defaultPath = "/tmp/version/"
+)
+
+//GetChartsPath is used to construct path for given chart.
+func GetChartsPath(ctx context.Context, chartsInput model.ChartsInput) string {
+	UserName := chartsInput.UserName
+	RepoName := chartsInput.RepoName
+	RepoBranch := chartsInput.RepoBranch
+	HubName := chartsInput.HubName
+	ChartsPath := defaultPath + UserName + "/" + HubName + "/" + RepoName + "/" + RepoBranch + "/charts/"
+	return ChartsPath
+}
+
+//GetExperimentPath is used to construct path for given experiment.
+func GetExperimentPath(ctx context.Context, experimentInput model.ExperimentInput) string {
+	UserName := experimentInput.UserName
+	RepoName := experimentInput.RepoName
+	RepoBranch := experimentInput.RepoBranch
+	HubName := experimentInput.HubName
+	experimentName := experimentInput.ExperimentName
+	chartName := experimentInput.ChartName
+	ExperimentPath := defaultPath + UserName + "/" + HubName + "/" + RepoName + "/" + RepoBranch + "/charts/" + chartName + "/" + experimentName + "/" + experimentName + ".chartserviceversion.yaml"
+	return ExperimentPath
```

Comment:
can we access the fields directly rather than reassigning them to a temporary variables which are not used elsewhere?

Reply:
@ishangupta-ds We did this intentionally for readability, the line for path will be larger in size, if we use dot operator directly in path for accessing different keys in struct.We don't have any problem in having like that if we need it that way.